backends/ubpf/p4include/ubpf_model.p4

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2019 Orange
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef _FILTER_MODEL_P4_
+#define _FILTER_MODEL_P4_
+
+#include <core.p4>
+
+parser parse<H, M>(packet_in packet, out H headers, inout M meta);
+control filter<H, M>(inout H headers, inout M meta);
+@deparser
+control deparser<H>(packet_out b, in H headers);
+
+package ubpf<H, M>(parse<H, M> prs,
+                filter<H, M> filt,
+                deparser<H> dprs);
+
+extern void mark_to_drop();
```

Comment:
I would suggest documenting these functions.

Reply:
In particular, for mark_to_drop() and mark_to_pass(), what is the default behavior after a packet is done processing?  Probably pass?

Then if the P4 program does mark_to_drop() and/or mark_to_pass() calls, whichever one was last determines whether the packet is passed or dropped later?

And there are no user-visible P4 variables that are modified by mark_to_drop() or mark_to_pass()?  I would not ask, except this was an issue for several years with v1model's mark_to_drop(), and why it was changed to take a parameter later.  It appears you have nothing that corresponds to standard_metadata, so it would make sense if in your architecture, mark_to_drop() and mark_to_pass() only modify state "hidden from the user's P4 program" inside the architecture implementation.

Another good thing to document about these extern functions is which of the architecture's parsers/controls they are allowed to be called in, e.g. maybe you intended something like "mark_to_drop() and mark_to_pass() may only be called in the filter control".

Comments in the p4include/v1model.p4 file would be good to look at for examples.  If any of them happen to match the behavior you provide (e.g. that might be true for your Register extern compared to v1model's register extern), feel free to copy and paste whatever is accurate for your architecture.